backend/src/middleware/handleHtmlContent/handleContentData.js

```diff
@@ -47,24 +65,32 @@ const updateHashtagsOfPost = async (postId, hashtags, context) => {
   session.close()
 }
 
-const handleContentData = async (resolve, root, args, context, resolveInfo) => {
-  // extract user ids before xss-middleware removes classes via the following "resolve" call
+const handleContentDataOfPost = async (resolve, root, args, context, resolveInfo) => {
   const idsOfMentionedUsers = extractMentionedUsers(args.content)
-  // extract tag (hashtag) ids before xss-middleware removes classes via the following "resolve" call
   const hashtags = extractHashtags(args.content)
 
-  // removes classes from the content
   const post = await resolve(root, args, context, resolveInfo)
 
-  await notify(post.id, idsOfMentionedUsers, context)
+  await notifyMentions('Post', post.id, idsOfMentionedUsers, context)
   await updateHashtagsOfPost(post.id, hashtags, context)
 
   return post
 }
 
+const handleContentDataOfComment = async (resolve, root, args, context, resolveInfo) => {
+  const idsOfMentionedUsers = extractMentionedUsers(args.content)
+  const comment = await resolve(root, args, context, resolveInfo)
+
+  await notifyMentions('Comment', comment.id, idsOfMentionedUsers, context)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-15T22:42:12Z" title="Friday, August 16th 2019, 12:42:12 am +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -56,15 +63,29 @@ const handleContentData = async (resolve, root, args, context, resolveInfo) => {
   // removes classes from the content
   const post = await resolve(root, args, context, resolveInfo)
 
-  await notify(post.id, idsOfMentionedUsers, context)
+  await notifyMentions('Post', post.id, idsOfMentionedUsers, context)
   await updateHashtagsOfPost(post.id, hashtags, context)
 
   return post
 }
 
+const handleContentDataOfComment = async (resolve, root, args, context, resolveInfo) => {
+  // extract user ids before xss-middleware removes classes via the following "resolve" call
+  const idsOfMentionedUsers = extractMentionedUsers(args.content)
+
+  // removes classes from the content
+  const comment = await resolve(root, args, context, resolveInfo)
+
+  await notifyMentions('Comment', comment.id, idsOfMentionedUsers, context)
+
+  return comment
+}
+
 export default {
   Mutation: {
-    CreatePost: handleContentData,
-    UpdatePost: handleContentData,
+    CreatePost: handleContentDataOfPost,
+    UpdatePost: handleContentDataOfPost,
+    CreateComment: handleContentDataOfComment,
+    // UpdateComment: handleContentDataOfComment,
```


I cannot notify a user anymore if I update my comment :cry: ?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-16T06:01:08Z" title="Friday, August 16th 2019, 8:01:08 am +02:00">Aug 16, 2019</time>_

---


this is fixed in a subsequent `PR`, although I guess it could be added here as well @roschaefer

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-16T07:02:52Z" title="Friday, August 16th 2019, 9:02:52 am +02:00">Aug 16, 2019</time>_

---


I forgot to remove this comment signs …
But did now. 🤗
You can smile again 😆